File: AoC2017Tests/Day03Test.fs
module Day03Test

open AoC2017
open NUnit.Framework
open Common.Extensions

[<TestCase("1","0")>] 
[<TestCase("12","3")>] 
[<TestCase("23","2")>] 
[<TestCase("1024","31")>]
let Part1 (input: string, expected : string) =
    let day = new Day03(input.ToList())
    let actual = day.SolvePart1()
    Assert.AreEqual(expected, actual)

[<Test>]
let Part2 () = 
    let day = new Day03("day03")
    let actual = day.SolvePart2()
    Assert.AreEqual("", actual)

File: AoC2017Tests/Day03Test.cs
namespace AoC2017Tests;

public class Day03Test
{


	[TestCase("1", "0")]
	[TestCase("12", "3")]
	[TestCase("23", "2")]
	[TestCase("1024", "31")]
	public async Task Part1(string input, string expected)
	{
		var day03 = new Day03(new List<string> { input });
		var actual = await day03.SolvePart1( );
		Assert.That(actual, Is.EqualTo(expected));
	}

	
}
